### PR TITLE
Fix binary partition values in Spark.

### PR DESCRIPTION
**Pull request by rdblue**
Creating an InternalRow for a partition incorrectly returned the value of `ByteBuffer.get`, which is the buffer itself, not the array.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
@rdblue could you, please, give an example when you experienced this issue? Will it be easy to add a test for it?

**Comment by rdblue** [comment]
I do need to add a test for this. It should be fairly easy. Any binary value used in an identity partition field will hit this when ready by Spark, if the column is projected.

**Comment by rdblue** [comment]
@aokolnychyi, I've updated this with a test to verify all Spark data types can be used as partition fields. Could you take another look to review?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestPartitionValues.java [hunk 1]
```diff
@@ -53,11 +62,28 @@
     };
   }
 
-  private static final Schema SCHEMA = new Schema(
+  protected static final Schema SUPPORTED_PRIMITIVES = new Schema(
```
Comment (aokolnychyi) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestPartitionValues.java]: nit: Why protected? Do we expect `TestPartitionValues` to be extended and used in another package?

Comment (rdblue) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestPartitionValues.java]: I copied this from an abstract test class. I'll fix it.

**Pull request merged**